File: core-service/src/main/java/io/hppn/unirag/service/TagService.java
package io.hppn.unirag.service;

import io.hppn.unirag.dto.tag.TagDTO;
import io.hppn.unirag.dto.tag.request.TagUpsertDTO;

import java.util.List;
import java.util.UUID;

public interface TagService {
    TagDTO getById(UUID id);

    List<TagDTO> getAll();

    TagDTO upsert(TagUpsertDTO dto);

    void delete(UUID id);
}

File: capstone-rag/src/services/tagService.ts
// tagService.ts
import { TagDTO, TagUpsertDTO } from "@/types/tag";
import apiClient from "./_client";

export const tagService = {
  getAll: async (): Promise<TagDTO[]> => {
    const response = await apiClient.get("/api/tags");
    return response.data;
  },

  getById: async (id: string): Promise<TagDTO> => {
    const response = await apiClient.get(`/api/tags/${id}`);
    return response.data;
  },

  create: async (data: TagUpsertDTO): Promise<TagDTO> => {
    const response = await apiClient.post("/api/tags", data);
    return response.data;
  },

  update: async (id: string, data: TagUpsertDTO): Promise<TagDTO> => {
    const response = await apiClient.put(`/api/tags/${id}`, data);
    return response.data;
  },

  delete: async (id: string): Promise<void> => {
    await apiClient.delete(`/api/tags/${id}`);
  },
};
